framework/include/interfaces/PostprocessorInterface.h

```diff
@@ -76,29 +76,44 @@ class PostprocessorInterface
    *
    * see getPostprocessorValue getPostprocessorValueOld getPostprocessorValueOlder
    */
-  const PostprocessorValue & getPostprocessorValueByName(const PostprocessorName & name) const;
+  virtual const PostprocessorValue &
+  getPostprocessorValueByName(const PostprocessorName & name) const;
   const PostprocessorValue & getPostprocessorValueOldByName(const PostprocessorName & name) const;
   const PostprocessorValue & getPostprocessorValueOlderByName(const PostprocessorName & name) const;
   ///@}
 
-  ///@{
   /**
-   * Return the default postprocessor value
-   * @param name The name of the postprocessor parameter
-   * @return A const reference to the default value
+   * Determine whether or not the Postprocessor is a default value. A default value is when
+   * the value is either the value set by addParam, or is a user-set value in input instead of
+   * a name to a postprocessor.
+   * @param param_name The name of the Postprocessor parameter
+   * @param index The index of the postprocessor
+   * @return True if the Postprocessor is a default value, false if the Postprocessor
+   * is the name of a Postprocessor
    */
-  const PostprocessorValue & getDefaultPostprocessorValue(const std::string & name) const;
-  ///@}
+  bool isDefaultPostprocessorValue(const std::string & param_name,
+                                   const unsigned int index = 0) const;
+
+  /**
+   * Get the default value associated with a Postprocessor parameter.
+   * You should check isDefaultPostprocessorValue to verify that the parameter
+   * is a default value before calling this method.
+   * @param param_name The name of the Postprocessor parameter
+   * @param index The index of the postprocessor
+   * @return The default value
+   */
+  PostprocessorValue getDefaultPostprocessorValue(const std::string & param_name,
```

Comment:
Was dropping the `const &` intentional?

Reply:
Yeah. This isn't a reference for people to store. If they want the value at all, they should just call getPostprocessorValue(), which will create something they can bind to if it's a default value. I think that I should actually just remove this method. I don't think it's used.

Reply:
I say remove it if you can.